pkg/update/update.go

```diff
@@ -0,0 +1,132 @@
+// Copyright Nitric Pty Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package update
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/pterm/pterm"
+	"github.com/spf13/cobra"
+
+	"github.com/nitrictech/cli/pkg/utils"
+)
+
+var (
+	shouldShowWarning bool
+	CurrentVersion    string
+	LatestVersion     string
+)
```

Comment:
Lets avoid using global state where we can.

The workflow for this could probably be:

1. FetchLatestVersion() - This will fetch and cache the latest version of nitric (once per day to avoid github API throttling). This result will be stored like is already happening. This method can probably just return the latest version from cache as well (and only refresh when stale).

2. `isOutdated()` - Can probably be private for now, will calculate if the version is out of date based on the cache from FetchLatestVersion.

3. `PrintOutdatedWarning()` - Prints the warning if `isOutdated() === true`.